lib/api/v3/work_packages/work_packages_by_project_api.rb

```diff
@@ -44,19 +44,50 @@ def create_service
                   project: @project,
                   send_notifications: !(params.has_key?(:notify) && params[:notify] == 'false'))
             end
+
+            def set_filters_from_json(query, json)
+              filters = JSON.parse(json)
+              operators = filters.inject({}) { |result, filter|
+                attribute = filter.keys.first # there should only be one attribute per filter
+                result[attribute] = filter[attribute]['operator']
+                result
+              }
+              values = filters.inject({}) { |result, filter|
+                attribute = filter.keys.first # there should only be one attribute per filter
+                result[attribute] = filter[attribute]['values']
+                result
+              }
+
+              query.filters = []
+              query.add_filters(filters.map(&:keys).flatten, operators, values)
+            end
+
+            def collection_representer(work_packages, filter_json:)
```

Comment:
Assignment Branch Condition size for collection_representer is too high. [16.31/15]